### PR TITLE
Feat/dark theme support

### PR DESCRIPTION
**Pull request by skjalgepalg**
# Add support for a default dark theme in core-docs

* opt-in via configurable options
```html
<script>
  window.coreDocs = {
    theme: { 
      label: 'dark mode', // theme switch label (default: 'dark mode')
      prefers: true // use system settings for light or dark mode (default: true)
    }
  }
</script>
```
* theme optionally can adhere to user preferences `prefers-color-scheme: dark` 
  * theme can be toggled by used and is saved in current session only
  * user toggled theme has higher presidency then system theme (toggle > prefers-color-scheme: dark)
* possible to override CSS custom properties to control themes
* migrate from deprecated [code-prettify](https://github.com/googlearchive/code-prettify) to [highlight.js](https://highlightjs.org/)
* use accessibility friendly code highlight themes [A 11 Y DARK/LIGHT](https://highlightjs.org/static/demo/)
* toggle custom classes in code examples and demos with conditional classes `class="{{ 'light' : 'dark' }}"`
* migrate to sass from css
* default options; start core docs without `window.coreDocs`
* extract styles to separate css file (browser seems to resolve it faster then when injected through js)

### CODE REVIEW

**Comment by skjalgepalg** [comment]
Screenshot 2022-10-24 at 08 34 22
Screenshot of docs-page in default dark theme with visible toggle

File: readme.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ core-docs can parse and render all markdown files you put in your repository. Th
 
 ```html
 <!doctype html>
+<head>
+  <meta charset="utf-8">
+  <link rel="stylesheet" href="https://static.nrk.no/core-docs/major/2/core-docs.css"/>
```
Comment (skjalgepalg) [inline readme.md:18]: Burde vi hatt en kommentar inline her vedr. innlasting tidlig for å unngå white-flashing ved navigasjon/reload?

Comment (klizter) [reply readme.md:18]: har gått tilbake til `import styles from './index.scss'`

**Comment by skjalgepalg** [comment]
Burde legge inn et notat i docs om at vi har tweaket noen av fargene i a11y-light for å fungere med noe mørkere bakgrunn

**Comment by klizter** [comment]
> Burde legge inn et notat i docs om at vi har tweaket noen av fargene i a11y-light for å fungere med noe mørkere bakgrunn

ordner det 👨‍🔧

File: package.json [hunk 2]
```diff
@@ -29,21 +39,24 @@
     "@rollup/plugin-replace": "3.0.1",
     "@rollup/plugin-terser": "0.1.0",
     "autoprefixer": "10.4.13",
-    "rollup": "2.79.1",
+    "postcss": "8.4.18",
+    "rollup-plugin-copy": "3.4.0",
     "rollup-plugin-postcss": "4.0.2",
     "rollup-plugin-serve": "2.0.2",
+    "rollup-plugin-terser": "7.0.2",
```
Comment (skjalgepalg) [inline package.json:46]: should be removed

File: src/index.scss [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+  &-wrapper {
+    $size: 1.8em;
+    font-size: 18px;
+    height: $size;
+    width: $size;
+    border-radius: 3em;
+
+    html[data-theme='light'] &,
+    html & {
+      border: 1px solid hsla(210, 15%, 50%, 0.4);
```
Comment (skjalgepalg) [inline src/index.scss:287]: Get color from shade variable and split into several shades

File: src/readme.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+</style>
+```
+
+The following CSS Custom properties are used for core-docs look and feel:
+
+```CSS
+:root {
+  /* Values for light/default theme */
+  --docs-color-background--light: #fff;
+  --docs-color-background-code--light: #fefefe;
```
Comment (skjalgepalg) [inline src/readme.md:253]: update to used value

**Pull request merged**